src/lib/pagination/view.coffee

```diff
@@ -0,0 +1,134 @@
+$ = require( 'jqlite' )
+EventsEmitter = require( 'events' )
+Pagination = require( './index' )
+PaginationUpdater = require( './updater' )
+
+###
+ * # GridPaginationView
+ * Builds a view for a Pagination instance.
+ *
+ * Buttons are created and then updated on every
+ * Pagination::CHANGE event.
+ *
+ * Keypress events are listenned to so the user
+ * can navigate through pagination with the keyboard.
+###
+class GridPaginationView extends EventsEmitter
+
+  TAGNAME: 'BUTTON'
+
+  CLICK: 'syn.grid.pagination.view.click'
+
+  PAGE_CLASS: '--pagination-page'
+  PREV_CLASS: '--pagination-prev'
+  NEXT_CLASS: '--pagination-next'
+
+  ###
+   * Emits a "page change" event
+   * @param  {Event} event
+   * @return {undefined}
+  ###
+  _buttonClickHandler: ( event ) =>
+    page = event.target.getAttribute( 'page' )
+    @emit( @CLICK, page ) unless !page
+
+  ###
+   * On "<" or ">", it simulates a click on prev or next button
+   * @param  {Event} event
+   * @return {undefined}
+  ###
+  _keyPressHandler: ( event ) =>
+    @prev?.trigger( 'click' ).focus() if event.keyCode is 37
+    @next?.trigger( 'click' ).focus() if event.keyCode is 39
+
+  ###
+   * @constructor
+   * @param  {Pagination} @pagination Pagination instance
+  ###
+  constructor: ( @pagination ) ->
+    @element = $( document.createElement( 'SPAN' ) )
+    @buttons = []
+    @prev = @next = null
+    @updater = new PaginationUpdater( @pagination, this )
+    @element.on( 'keydown', @_keyPressHandler )
+
+    @length( @pagination.buttons() )
+
+  ###
+   * Define length of nav bar
```

Comment:
Defines/gets
Many functions in the commit have been designed as getters/setters.